pkg/controller/migmigration/dvm.go

```diff
@@ -104,29 +105,48 @@ func (t *Task) hasDirectVolumeMigrationCompleted(dvm *migapi.DirectVolumeMigrati
 	switch {
 	case dvm.Status.Phase != "" && dvm.Status.Phase != dvmc.Completed:
 		// TODO: Update this to check on the associated dvmp resources and build up a progress indicator back to
-		progress = append(progress, fmt.Sprintf("direct volume migration started at %v. ", dvm.Status.StartTimestamp))
-		progress = append(progress, t.getDVMProgress(dvm.Status.RunningPods)...)
 	case dvm.Status.Phase == dvmc.Completed && dvm.Status.Itinerary == "VolumeMigration":
-		progress = append(progress, fmt.Sprintf("%v/%v volume migrations were successful", successfulPods, totalVolumes))
 		completed = true
 	case (dvm.Status.Phase == dvmc.MigrationFailed || dvm.Status.Phase == dvmc.Completed) && dvm.Status.Itinerary == "VolumeMigrationFailed":
 		failureReasons = append(failureReasons, fmt.Sprintf("direct volume migration failed. %s", volumeProgress))
 		completed = true
 	default:
 		progress = append(progress, volumeProgress)
 	}
+	progress = append(progress, t.getDVMPodProgress(dvm.Status.RunningPods, "Running")...)
+	progress = append(progress, t.getDVMPodProgress(dvm.Status.SuccessfulPods, "Completed")...)
+	progress = append(progress, t.getDVMPodProgress(dvm.Status.FailedPods, "Failed")...)
+
+	// sort the progress report so we dont have flapping for the same progress info
+	sort.Sort(sort.StringSlice(progress))
```

Comment:
Just curious on this one, has progress info order flapping been observed? I would expect an ordered slice to stay ordered.

Reply:
no flapping has not been observed, this was carried over from previous work where the order of underlying progress was not guaranteed, but now it seems we can remove the sort